OpenSSL/SSL.py

```diff
@@ -310,6 +311,17 @@ def load_verify_locations(self, cafile, capath=None):
         :param capath: In which directory we can find the certificates
         :return: None
         """
+
+        # Backward compatibility
+        if isinstance(cafile, _text_type):
+            if version_info.major == 2:
+                warn("unicode in cafile is no longer accepted, use bytes", DeprecationWarning)
+            elif version_info.major == 3:
+                warn("str in cafile is no longer accepted, use bytes", DeprecationWarning)
+            else:
+                warn("text in cafile is no longer accepted, use bytes", DeprecationWarning)
```

Comment:
You can assume that the only versions of Python this code needs to handle are 2 and 3.  Also you might find `six.PY3` a more convenient spelling of the version check.